server/routes.js

```diff
@@ -3,42 +3,55 @@ const restify = require("restify");
 const Plus = require("./plus-model");
 const { hash } = require("./util");
 const { getEmbedScript } = require("./embed");
+const {
+  catchAsyncErrors,
+  decodeLocation,
+  assertOriginMatchesLocation
+} = require("./middleware");
+const log = require("./logger")("routes");
+
+const THROTTLE_MODE = process.env.THROTTLE_MODE || "ip";
+const THROTTLE_RATE = Number(process.env.THROTTLE_RATE) || 0.5;
+log.debug(`THROTTLE_MODE=${THROTTLE_MODE}`);
+log.debug(`THROTTLE_RATE=${THROTTLE_RATE}`);
 
 function createRoutes(server) {
   server.get("/", (req, res, next) => {
     res.send({ version, name });
     return next();
   });
 
-  server.get("/plus/:location", async function(req, res, next) {
-    const { count, rows } = await Plus.findAndCountAll({
-      where: {
-        location: hash(decodeURIComponent(req.params.location))
-      }
-    });
-    res.send({
-      count,
-      latest: rows[rows.length - 1]
-    });
-    return next();
-  });
+  server.get(
+    "/plus/:location",
+    decodeLocation,
+    catchAsyncErrors(async function(req, res, next) {
+      const { location } = req.params;
+      log.debug(`Fetching count for ${location}`);
+      const { count, rows } = await Plus.findAndCountAll({
+        where: { location: hash(location) }
+      });
+      res.send({
+        count,
+        latest: rows[rows.length - 1]
+      });
+      next();
+    })
+  );
 
   server.post(
     "/plus",
-    restify.plugins.throttle({ burst: 10, rate: 1, xff: true }),
-    function(req, res, next) {
-      return Plus.create({
-        location: decodeURIComponent(req.body.location)
-      })
-        .then(resp => {
-          res.send(resp);
-          next();
-        })
-        .catch(err => {
-          res.send(err);
-          next();
-        });
-    }
+    restify.plugins.throttle({
```

Comment:
what does the throttle do? too lazy to look it up :/

Reply:
prevents some spamming. default limits requests per ip address to 1 per 2 seconds. you can adjust the rate with the env variable.